examples/example-gateway/clients/baz/baz.go

```diff
@@ -0,0 +1,215 @@
+// TODO: (lu) generate
+
+package bazClient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/uber/zanzibar/runtime"
+
+	zt "github.com/uber/zanzibar/runtime/tchannel"
+	"go.uber.org/thriftrw/wire"
+
+	"github.com/uber/zanzibar/examples/example-gateway/build/gen-code/github.com/uber/zanzibar/clients/baz/baz"
+)
+
+// TChanBaz is the interface that defines the server handler and client interface.
+type TChanBaz interface {
+	Call(ctx context.Context, reqHeaders map[string]string, r *baz.SimpleService_Call_Args) (map[string]string, *baz.BazResponse, error)
+	Simple(ctx context.Context, reqHeaders map[string]string) (map[string]string, error)
+	SimpleFuture(ctx context.Context, reqHeaders map[string]string) (map[string]string, error)
+}
+
+// NewClient returns a new http client for service Bar.
+func NewClient(config *zanzibar.StaticConfig, gateway *zanzibar.Gateway) *BazClient {
+	// TODO: (lu) get serviceName from thrift
+	serviceName := "SimpleService"
+
+	ch := gateway.Channel
+
+	ip := config.MustGetString("clients.baz.ip")
+	port := config.MustGetInt("clients.baz.port")
+	ch.Peers().Add(ip + ":" + strconv.Itoa(int(port)))
```

Comment:
We cannot mutate the `Peers()` of the top channel.

We need to allocate a sub channel.